integration-testing/src/debugAgentTest/kotlin/PrecompiledDebugProbesTest.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2016-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+import org.junit.Test
+import kotlin.test.*
+
+/*
+ * This is intentionally put here instead of coreAgentTest to avoid accidental classpath replacing
+ * and ruining core agent test.
+ */
+class PrecompiledDebugProbesTest {
+
+    @Test
+    fun testClassFileContent() {
+        val clz = Class.forName("kotlin.coroutines.jvm.internal.DebugProbesKt")
+        val className: String = clz.getName()
+        val classFileResourcePath = className.replace(".", "/") + ".class"
+        val stream = clz.classLoader.getResourceAsStream(classFileResourcePath)!!
+        val array = stream.readBytes()
+        val binFile = clz.classLoader.getResourceAsStream("DebugProbesKt.bin")!!
+        val binContent = binFile.readBytes()
+        assertTrue(array.contentEquals(binContent))
```

Comment:
It should include some helpful message like `Blah-blah binary does not match, please copy this file to there`. Note that this will have to be done with every update of kotlin compiler version, since the compiler version is stored in the compiled metadata, so this process needs some automation (some kind of "overwrite" switch).